tools/shared.py

```diff
@@ -49,6 +49,27 @@ def exit_with_error(msg, *args):
   sys.exit(1)
 
 
+def fix_import_name(g):
+  if g.startswith('Math_'):
+    return g.split('_')[1]
+
+  # Side modules import from the main module, which still export
+  # using a prefixed '_', so don't change that
+  if not Settings.WASM_BACKEND and (Settings.MAIN_MODULE or Settings.SIDE_MODULE):
+    return g
+
+  if g.startswith('_'):
+    return g[1:]
+  # NB: these are imported directly into ASMJS modules
+  #     (see Settings.RUNTIME_FUNCS_TO_IMPORT)
+  #     and used by fastcomp, but their import name conflicts
+  #     with the library.js version (which is the canonical version),
+  #     so import them under a different name
+  if not Settings.WASM_BACKEND and g in ["abort", "getTempRet0", "setTempRet0"]:
+    return '__runtime_' + g
```

Comment:
I'm a little confused about this.    Do these function end up implement in the asm.js module itself?  If so then surely they wouldn't actually be imported at all?

Are any of these __runtime_ functions actually implemented?   Or does this prefix get stripped off somewhere else (I don't see it anywhere else).

Reply:
They get imported using "_runtime" prefix but used without the runtime prefix. e.g.:

    var setTempRet0 = env.__runtime_setTempRet0;

The library.js versions of those functions are used with the "_" prefix but imported without the "_" prefix:

    var _setTempRet0 = env.setTempRet0;

Previously these were imported as in the following:

    var setTempRet0 = env.setTempRet0;
    var _setTempRet0 = env._setTempRet0;

Reply:
This part of the change still feels strange.  I don't understand why we would need or want two versions of these symbols.   Anyway,  perhaps @kripken 's suggestion to remove the second copy would mean that this part was no longer needed?